### DISCUSSION TITLE
Unable to restore arch id mappings for 'archname'

### DISCUSSION THREAD

**Discussion by sprout42** [Vector35/binaryninja-api#3022] [Architectures] [answered]
When writing my own architecture plugin I always get this error when Binja starts up and as far as I can tell it should be working correctly.  I've seen this error over multiple versions.  This error was taken from the latest development version `3.0.3298-dev`:
```
[Default] No bundled types for platform archname found in system types
[Default] Unable to restore arch id mappings for 'archname'
[Default] Unable to restore arch id mappings for 'archname'
```

**Comment by fuzyll** [comment, marked as answer]
What version was the original database you've loaded saved on? That error message is expected (and not an issue) *if*:

1. You're using a custom architecture
2. The database is from an old Binary Ninja version from about 2-3 years ago

If those apply to you, you can safely ignore the errors and you should stop seeing them if/when you upgrade that database.

If those *don't* apply to you, we might need to look into it further because my understanding (from looking at the relevant code) is that analysis should never head down that codepath.

**Reply by sprout42** [reply]
This was for a custom architecture.  but I see it when I load up binary ninja without loading any database.  it seems related to just loading my custom architecture plugin.

**Reply by fuzyll** [reply]
That's...pretty strange. Have you created any type libraries for your custom architecture? That codepath is also used while loading type libraries, so that's the only other thing I can think of that would trigger the error message. Does `arch.type_libraries` have something in it?

**Reply by rssor** [reply]
Are you getting the message as soon the window appears before you've opened any files at all? It's related to deserialization of either types or a database, neither of which I'd expect prior to interaction of some sort after start.

**Reply by sprout42** [reply]
no type libraries have been created for this arch
```
>>> bv.arch.type_libraries
[]
```

**Reply by sprout42** [reply]
And yes, I get this as soon as the GUI has opened before I open any files.

**Reply by sprout42** [reply]
Correction! I just checked with the latest build of Binary Ninja (3.0.3301-dev) and it does not show up by default, only after opening up a database.  I may have been mistaken before.

**Reply by fuzyll** [reply]
Hey, sorry, it looks like we kinda dropped this accidentally.

So, to confirm, this message shows up for you *only* when you open up a database that uses your custom architecture plugin? Or is it any database while your plugin is enabled? Or..?

Does it still happen on the latest version of `dev` (3.0.3322-dev)?

And you can confirm that, when it does show up, you have no type libraries in `bv.arch.type_libraries`?

I think this is probably a bug, so if you're able to confirm a bunch of those things for me, I'll go ahead and create an issue out of this discussion.

**Reply by sprout42** [reply]
Correct, it does only show up when I open a bndb (or analyze a new binary) with my architecture plugin.

I just tested this morning on version 3.0.3329-dev.

and there are no type libraries installed
```
>>> bv.arch.type_libraries
[]
```
